lib/tic_tac_toe.ex

```diff
@@ -26,9 +32,9 @@ defmodule TicTacToe do
          %{players: [first_player, second_player], board: board} = state
        ) do
     first_player = Player.selection(first_player, board)
+    {selection, token} = Player.get(first_player)
 
-    board =
-      BoardUpdate.place_token(board, first_player.selection, first_player.token)
+    board = BoardUpdate.place_token(board, selection, token)
```

Comment:
This is a little curious to me. `{selection, token}` are return from `Player.get` and then used together as arguments to `BoardUpdate.place_token`. It might be interesting to explore pushing the player down into `BoardUpdate`.

```
board = BoardUpdate.place_token(board, player)
```

I am not sure that is the right move, but maybe worth exploring. 

I think the gist of my curiosity here is that we have two values that are being returned together, and then being used independently as arguments to another function.

An alternative to pushing `Player` down into `BoardUpdate` might be to rethink the return value(s) of `Player.get` and the arguments to `.place_token` as a **single value** instead of two different values.

Reply:
I'll rethink.  I changed because initially I was getting the value from the data structure so wanted a getter method.

Reply:
Also, I want to add that it is conventional to return tuples in Elixir, but the convention is to return a tuple with an atom that serves as a tag followed by a value. This makes it easier to do pattern matching (similar to the `BoardInspect.outcome` function) and it also encodes some meaning.

```
{:ok, success_value}
{:error, reason}
{:win, winner}
{:draw, _}
```

When tuple just has values the meaning is implicit.
```
{1, "X"}
{3, 9} # hard to say what this means. Is 3 the selection, and 9 is the token? Or is this side size and board length? 😬 
```

Reply:
I wanted to offer that to piggyback off of @paytonrules comment on returning tuples!

Reply:
I see.  Now makes a bit more sense.